File: schemaorg/locationfeaturespecification.go
package schemaorg

import "encoding/json"

// LocationFeatureSpecification see : https://schema.org/LocationFeatureSpecification
type LocationFeatureSpecification struct {

PropertyValue

typeContext

// HoursAvailable see : https://schema.org/hoursAvailable
// The hours during which this service or contact is available.
HoursAvailable *OpeningHoursSpecification `json:"hoursAvailable"`

// ValidFrom see : https://schema.org/validFrom
// The date when the item becomes valid.
ValidFrom interface{} `json:"validFrom"`

// ValidThrough see : https://schema.org/validThrough
// The date after when the item is not valid. For example the end of an offer, salary period, or a period of opening hours.
ValidThrough interface{} `json:"validThrough"`

}

func (v *LocationFeatureSpecification) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "LocationFeatureSpecification"

	return json.Marshal(v)
}

File: schemaorgphp/classes/location_feature_specification.php
<?php

class LocationFeatureSpecification extends PropertyValue implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'LocationFeatureSpecification';
	
	/**
	 * The hours during which this service or contact is available.
	 * see : https://schema.org/hoursAvailable
	 * @var \OpeningHoursSpecification|\OpeningHoursSpecification[]
	 */
	public var $hours_available;
	
	/**
	 * The date when the item becomes valid.
	 * see : https://schema.org/validFrom
	 * @var string|string[]
	 */
	public var $valid_from;
	
	/**
	 * The date after when the item is not valid. For example the end of an offer, salary period, or a period of opening hours.
	 * see : https://schema.org/validThrough
	 * @var string|string[]
	 */
	public var $valid_through;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'LocationFeatureSpecification'
		);
		
		$serialized = so_json_serialize( $this->hours_available );
		if ( ! empty( $serialized ) ) {
			$out['hoursAvailable'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->valid_from );
		if ( ! empty( $serialized ) ) {
			$out['validFrom'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->valid_through );
		if ( ! empty( $serialized ) ) {
			$out['validThrough'] = $serialized;
		}
		
		return $out;
	}
}
